### ISSUE TITLE
my core.chatcmd_parse_coords functionset proposal

### ISSUE THREAD

**Issue by imre84** [minetest/minetest#13592]
Hello,

so the issue I'm having: as an administrator you can never be sure whether the chatcommand you're about to use accepts coordinates in x y z or x,y,z or (x,y,z) format. Or whether tilde notation is accepted.
as a mod developer you might want to develop a /myawesomechatcmd \<pos1> \<pos2> \<somerandomparam3> command, and unless you want to have a restrictive format for the positions it would be hard to parse your command arguments

so my vision here is to provide the following core.chatcmd_parse_* routines in order to fix all the above issues: it supports all 3 of the coordinate formats, tilde notation, and returns with the unparsed (2nd and 3rd parameters) portion of the param string (so that you can call core.chatcmd_parse_* on the rest). Also in case of a parse error, it can give you a pretty good idea of what went wrong, and approximately where

here is an out-of-the box working demo of what I'm thinking
https://github.com/imre84/chatcmd_parse_coords
the guys at the IRC channel said I should be starting with an issue before I revamp your core completely
so let me know how to move forward from this point. Thank you in advance.

**Comment by SmallJoker** [comment]
I think it would be better to make `core.parse_coordinates` accept a string as argument as well because currently this function requires additional "regex" boilerplate which could as well be integrated directly into this function.

https://github.com/minetest/minetest/blob/206198431309fb00ff717f3ebd5e7c0af3de59dd/builtin/game/chat.lua#L884-L886

**Comment by imre84** [comment]
> EDIT: However, this will not solve your issue with mods behaving differently. The best option there is to ask the authors to use a "more common" notation which is yet to be clarified.

my solution would be to let all notations coexist as per issue
